desktop-widgets/tableview.h

```diff
@@ -45,7 +47,7 @@ class TableView : public QGroupBox {
 
 private:
 	Ui::TableView ui;
-	QPushButton *plusBtn;
+	QPushButton *plusBtn, *mirrorBtn;
```

Comment:
Now that looks like a layering violation to me. Why would the general `TableView` class have a "mirror button"!?

Reply:
I would prefer if this was addressed (or at least there was a response to the comment that explained why Berthold and I are wrong here)

Reply:
Hmm. Adding the button only in the planner is painful (you need to tell it where to go), in particular due to the non-standard positioning (top right) that needs to keep adjusting when the table is resized.

If I understand correctly, the proper way to handle this would be to derive a class PlannerTableView from TableView that has this additional button. Do you really want me to do this (rather than this hack of an invisible button that only becomes visible when the TableView is used in the planner)?

Reply:
I am currently looking into adding an 'Actions' menu that then can contain less often used functionality that is currently hard to find because it is behind keyboard shortcuts and context menus.
I think this will be a good candidate for this 'Actions' menu, which will remove the need for a button and subclassing of TableView.

Reply:
> If I understand correctly, the proper way to handle this would be to derive a class PlannerTableView from TableView that has this additional button. Do you really want me to do this (rather than this hack of an invisible button that only becomes visible when the TableView is used in the planner)?

Either that or add a general `addButton` function that takes an icon and a callback. I can do this, if you want.

That said, I am not a fan of the whole `TableView` class at all.